src/main/scala/com/wavesplatform/network/RemoteScoreObserver.scala

```diff
@@ -11,105 +11,96 @@ import scorex.utils.ScorexLogging
 
 import scala.concurrent.duration.FiniteDuration
 
-
 @Sharable
 class RemoteScoreObserver(scoreTtl: FiniteDuration, lastSignatures: => Seq[ByteStr], initialLocalScore: BigInt)
   extends ChannelDuplexHandler with ScorexLogging {
 
-  private val pinnedChannel = new AtomicReference[Channel]()
-
-  @volatile
-  private var localScore = initialLocalScore
+  private type ScorePair = (Channel, BigInt)
 
   private val scores = new ConcurrentHashMap[Channel, BigInt]
 
-  private def channelWithHighestScore =
+  @volatile private var localScore = initialLocalScore
+  private val currentRequest = new AtomicReference[Option[ScorePair]](None)
```

Comment:
Pinned channel score doesn't seem to be used anywhere, so why not just have a reference to `Option[Channel]`? That way the code may be simplified even further.